bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -129,6 +135,103 @@ enum ClusterConfig {
     },
 }
 
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Config {
+    api_version: String,
+    kind: String,
+    metadata: Metadata,
+    kubernetes_network_config: KubernetesNetworkConfig,
+    addons: Vec<Addon>,
+    iam: IAMConfig,
+    managed_node_groups: Vec<ManagedNodeGroup>,
+}
+
+#[derive(Serialize)]
+struct Metadata {
+    name: String,
+    region: String,
+    version: String,
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct KubernetesNetworkConfig {
+    ip_family: String,
+}
+
+#[derive(Serialize)]
+struct Addon {
+    name: String,
+    version: String,
+}
+
+#[derive(Serialize)]
+#[allow(non_snake_case)]
+struct IAMConfig {
+    withOIDC: bool,
+}
+
+#[derive(Serialize)]
+struct ManagedNodeGroup {
+    name: String,
+}
+
+fn create_yaml(cluster_name: &str, region: &str, version: &str) -> ProviderResult<()> {
+    let ip_family = if cluster_name.ends_with("ipv6") {
+        IPV6
+    } else {
+        IPV4
+    };
+
+    let cluster = Config {
+        api_version: "eksctl.io/v1alpha5".to_string(),
+        kind: "ClusterConfig".to_string(),
+        metadata: Metadata {
+            name: cluster_name.to_string(),
+            region: region.to_string(),
+            version: version.to_string(),
+        },
+        kubernetes_network_config: KubernetesNetworkConfig {
+            ip_family: ip_family.to_string(),
+        },
+        addons: vec![
+            Addon {
+                name: "vpc-cni".to_string(),
+                version: "latest".to_string(),
+            },
+            Addon {
+                name: "coredns".to_string(),
+                version: "latest".to_string(),
+            },
+            Addon {
+                name: "kube-proxy".to_string(),
+                version: "latest".to_string(),
+            },
+        ],
+        iam: IAMConfig { withOIDC: true },
+        managed_node_groups: vec![ManagedNodeGroup {
+            name: "mng-1".to_string(),
+        }],
+    };
+
+    let yaml = serde_yaml::to_string(&cluster).expect("Failed to serialize YAML");
+
+    let mut file = File::create(CLUSTER_CONFIG_PATH)
```

Comment:
Do repeated runs use the same path? Will `create()` handle running this code a second time? What if this file wasn't put here by the code? Does this break any abilities to run multiple runs of testsys at the same time? Is there a reason we want this around after the fact or could we sidestep these concerns by using a temporary file location for the run?